src/main/java/org/broadinstitute/hellbender/utils/SequenceDictionaryUtils.java

```diff
@@ -0,0 +1,472 @@
+     * @param that
+     * @return
+     */
+    private static boolean sequenceRecordsAreEquivalent(final SAMSequenceRecord me, final SAMSequenceRecord that) {
+        if (me == that) return true;
+        if (that == null) return false;
+
+        if (me.getSequenceLength() != 0 && that.getSequenceLength() != 0 && me.getSequenceLength() != that.getSequenceLength())
+            return false;
+
+        // todo -- reenable if we want to be really strict here
+//        if (me.getExtendedAttribute(SAMSequenceRecord.MD5_TAG) != null && that.getExtendedAttribute(SAMSequenceRecord.MD5_TAG) != null) {
+//            final BigInteger thisMd5 = new BigInteger((String)me.getExtendedAttribute(SAMSequenceRecord.MD5_TAG), 16);
+//            final BigInteger thatMd5 = new BigInteger((String)that.getExtendedAttribute(SAMSequenceRecord.MD5_TAG), 16);
+//            if (!thisMd5.equals(thatMd5)) {
+//                return false;
+//            }
+//        }
+//        else {
+        if (me.getSequenceName() != that.getSequenceName())
+            return false; // Compare using == since we intern() the Strings
+//        }
+
+        return true;
+    }
+
+    /**
+     * A very simple (and naive) algorithm to determine (1) if the dict is a human reference (hg18, hg19, b36, or b37) and if it's
+     * lexicographically sorted.  Works by matching lengths of the static chr1, chr10, and chr2, and then if these
+     * are all matched, requiring that the order be chr1, chr2, chr10.
+     *
+     * @param dict
+     * @return
+     */
+    private static boolean nonCanonicalHumanContigOrder(SAMSequenceDictionary dict) {
+        if ( ! ENABLE_LEXICOGRAPHIC_REQUIREMENT_FOR_HUMAN ) // if we don't want to enable this test, just return false
+            return false;
+
+        SAMSequenceRecord chr1 = null, chr2 = null, chr10 = null;
+        for ( SAMSequenceRecord elt : dict.getSequences() ) {
+            if ( isHumanSeqRecord(elt, CHR1_HG18, CHR1_HG19, CHR1_B36, CHR1_B37) ) chr1 = elt;
+            if ( isHumanSeqRecord(elt, CHR2_HG18, CHR2_HG19, CHR2_B36, CHR2_B37) ) chr2 = elt;
+            if ( isHumanSeqRecord(elt, CHR10_HG18, CHR10_HG19, CHR10_B36, CHR10_B37) ) chr10 = elt;
+        }
+        if ( chr1 != null  && chr2 != null && chr10 != null) {
+            return ! ( chr1.getSequenceIndex() < chr2.getSequenceIndex() && chr2.getSequenceIndex() < chr10.getSequenceIndex() );
+        }
+
+        return false;
+    }
+
+    /**
+     * Trivial helper that returns true if elt has the same length as rec1 or rec2
+     * @param elt record to test
+     * @param recs the list of records to check for length equivalence
+     * @return true if elt has the same length as any of the recs
+     */
+    private static boolean isHumanSeqRecord(SAMSequenceRecord elt, SAMSequenceRecord... recs) {
+        for (SAMSequenceRecord rec : recs) {
+            if (elt.getSequenceLength() == rec.getSequenceLength() && elt.getSequenceName() == rec.getSequenceName()) {
```

Comment:
`elt.getSequenceName().equals(rec.getSequenceName())`, not `elt.getSequenceName() == rec.getSequenceName()` (Need to use `equals()` rather than `==` when comparing objects like Strings)